benchmarks/MOABB/utils/dataio_iterators.py

```diff
@@ -424,6 +402,80 @@ def get_tail_path(self, subject: int, session: Optional[str]) -> str:
         return tail_path
 
 
+class AsGraph(DataLoaderFactory):
+    def __init__(self, inner: DataLoaderFactory):
+        self.inner = inner
+
+    def prepare(
+        self,
+        *,
+        data_folder: str,
+        cached_data_folder: str,
+        dataset: MOABBDataset,
+        batch_size: int,
+        valid_ratio: float,
+        original_sample_rate: int,
+        target_subject_idx: int,
+        target_session_idx: Optional[int] = None,
+        sample_rate: Optional[int] = None,
+        fmin: Optional[float] = None,
+        fmax: Optional[float] = None,
+        tmin: Optional[float] = None,
+        tmax: Optional[float] = None,
+        n_steps_channel_selection: Optional[int] = None,
+        events_to_load: Optional[List[str]] = None,
+        save_prepared_dataset: bool = True,
+    ) -> Tuple[str, _SplitDataloaders]:
+        tail_path, datasets = self.inner.prepare(
+            data_folder=data_folder,
+            cached_data_folder=cached_data_folder,
+            dataset=dataset,
+            batch_size=batch_size,
+            valid_ratio=valid_ratio,
+            original_sample_rate=original_sample_rate,
+            target_subject_idx=target_subject_idx,
+            target_session_idx=target_session_idx,
+            sample_rate=sample_rate,
+            fmin=fmin,
+            fmax=fmax,
+            tmin=tmin,
+            tmax=tmax,
+            n_steps_channel_selection=n_steps_channel_selection,
+            events_to_load=events_to_load,
+            save_prepared_dataset=save_prepared_dataset,
+        )
+
+        edge_list = adjacency_mtx_to_edge_list(datasets["adjacency_mtx"])
+
+        for key in ("train", "valid", "test"):
+            datasets[key] = self.make_graph_dataloader(
+                datasets[key],
+                edge_list=edge_list,
+                ch_positions=torch.from_numpy(datasets["ch_positions"]),
+                shuffle=(key == "train"),
+            )
+
+        return tail_path, datasets
+
+    def make_graph_dataloader(
+        self,
+        dataloader: DataLoader,
+        edge_list: torch.Tensor,
+        ch_positions: torch.Tensor,
+        shuffle: bool = False,
+    ) -> torch_geometric.loader.DataLoader:
+        dataset = [
+            torch_geometric.data.Data(x=x, y=y, edge_index=edge_list)
+            for x, y in dataloader.dataset
+        ]
```

Comment:
@Drew-Wagner I think for the graph features shall we reshape the data here and have it like this as x: (C, T). We can also do this in the forward if you think is more better :)